quantum/process_keycode/process_tap_dance.h

```diff
@@ -41,6 +41,7 @@ typedef struct {
         tap_dance_user_fn_t on_each_tap;
         tap_dance_user_fn_t on_dance_finished;
         tap_dance_user_fn_t on_reset;
+        tap_dance_user_fn_t on_each_release;
```

Comment:
I added this at the end so existing user code that uses positional initialization doesn't break.  For example, the following will still compile even though only three of the four arguments are provided:

```
{ .fn = {user_fn_on_each_tap, user_fn_on_dance_finished, user_fn_on_dance_reset}, ... }
```